### ISSUE TITLE
Add support for IAsyncEnumerable element assertion

### ISSUE THREAD

**Issue by dallmair** [xunit/xunit#2367]
It would be great if xunit had dedicated support for asserting elements of `IAsyncEnumerable<T>` instances. Right now, we can of course transform an `IAsyncEnumerable<T>` to an `IEnumerable<T>` with any other means (e.g. the System.Linq.Async package), but native support in xunit would be nice.

Example API:

```
ValueTask Assert.EqualAsync<T>(IEnumerable<T> expected, IAsyncEnumerable<T> actual);
ValueTask Assert.EqualAsync<T>(IEnumerable<T> expected, IAsyncEnumerable<T> actual, IEqualityComparer<T> comparer);
```

`expected` will be in memory in most cases, so I think `IEnumerable<T>` is fine there.

**Comment by bradwilson** [comment]
I did a full pass, and decided to add both first parameter `IEnumerable<T>` as well as `IAsyncEnumerable<T>` for `Assert.Equal` and `Assert.NotEqual` for completeness.

Equal: https://github.com/xunit/assert.xunit/blob/22c89b0ea18d3a935352602201a2d8e6fa8d1158/AsyncCollectionAsserts.cs#L203-L311
NotEqual: https://github.com/xunit/assert.xunit/blob/22c89b0ea18d3a935352602201a2d8e6fa8d1158/AsyncCollectionAsserts.cs#L322-L430

For distribution, these will only show up when using `net6.0` or later (as that's the target framework where these would apply, as they're not supported on .NET Standard 1.1 [v2] or 2.0 [v3]). Since we didn't yet have a `net6.0` distribution for `xunit.v3.assert`, I added that as well.

**Comment by bradwilson** [comment]
Available in v2: `2.6.7-pre.3`
Available in v3: `0.1.1-pre.350`

https://xunit.net/docs/using-ci-builds

(I will likely bump the final v2 version to `2.7.0` because of the new assertions.)

**Comment by bradwilson** [comment]
Note, I discovered an issue with the source package (`xunit.assert.source`), which will be fixed in `2.6.7-pre.4` and `0.1.1-pre.351`. If you don't use the source package, the above builds work as expected.
